project/src/com/google/daggerqueryui/scripts/binding-graph.js

```diff
@@ -319,33 +319,50 @@ const bindingGraph = (function() {
     });
   }
 
+  /**
+   * Takes a string representing a path where two adjacent nodes are connected by an edge,
+   * and adds all these edges to the subgraph.
+   *
+   * <p>Given path is separated with ' -> ' arrow.
+   *
+   * <p>If the path is invalid and contains less than two nodes, the method does nothing.
+   *
+   * @param {string} path
+   * @param {string} baseColor
+   * @param {string} highlightColor
+   */
+  function addPath(path, baseColor, highlightColor) {
+    const nodesInPath = path.split( ' -> ');
+    if (nodesInPath.length < 2) {
+      return;
+    }
+
+    addEdge(nodesInPath[0], nodesInPath[1], new EdgeStyle(EdgeType.highlightSource, baseColor, highlightColor));
+    for (let index = 1; index + 2 < nodesInPath.length; ++index) {
+      addEdge(nodesInPath[index], nodesInPath[index + 1], new EdgeStyle(EdgeType.noHighlight, baseColor, highlightColor));
+    }
+    addEdge(nodesInPath[nodesInPath.length - 2], nodesInPath[nodesInPath.length - 1],
+      new EdgeStyle(EdgeType.highlightTarget, baseColor, highlightColor));
+  }
+
   return {
+
+
     /**
-     * Takes a string representing a path where two adjacent nodes are connected by an edge,
-     * and adds all these edges to the subgraph.
+     * Takes an array with paths and draws all of them.
      *
-     * <p>Given path is separated with ' -> ' arrow.
+     * <p>All given paths are separated with ' -> ' arrow.
      *
-     * <p>If the path is invalid and contains less than two nodes, the method does nothing.
-     *
-     * @param {string} path a valid string representing a path with at least two nodes
+     * @param {string[]} paths an array with valid paths between nodes in the graph
      */
-    addPath: function (path) {
-      const nodesInPath = path.split( ' -> ');
-      if (nodesInPath.length < 2) {
-        return;
-      }
-
+    addPaths: function (paths) {
       recolourAllNodes(GRAY_COLOR);
```

Comment:
Is this line still needed? It looks like you might already be setting the colour of all nodes in the addPath method.

Reply:
Oh, yes, you are right. It is an unnecessary action, since the graph does not contain any nodes at this moment.
Removed ✅